src/Microsoft.Data/DataFrame.cs

```diff
@@ -286,6 +284,59 @@ public IList<IList<object>> Tail(int numberOfRows)
         // TODO: Add strongly typed versions of these APIs
         #endregion
 
+        private DataFrame Clone(BaseColumn mapIndices = null, bool invertMapIndices = false)
+        {
+            List<BaseColumn> newColumns = new List<BaseColumn>(ColumnCount);
+            for (int i = 0; i < ColumnCount; i++)
+            {
+                newColumns.Add(Column(i).Clone(mapIndices, invertMapIndices));
+            }
+            return new DataFrame(newColumns);
+        }
+
+        public void SetColumnName(BaseColumn column, string newName) => _table.SetColumnName(column, newName);
+
+        public DataFrame Description()
+        {
+            bool IsNonNumericColumn(BaseColumn column)
```

Comment:
This isn't how this should work. What happens if someone wants to make an "Image" column in the future?

Instead, the column should be responsible for telling the DataFrame what it can and can't do.